utils/iterator/masked.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package iterator
+
+var _ Iterator[any] = (*masked[any])(nil)
+
+type masked[T any] struct {
+	parentIterator Iterator[T]
+	filter         func(T) bool
+}
+
+// NewMasked returns a new iterator that skips the elements in [parentIterator]
+// that if the element is in [filter].
+func NewMasked[T any](parentIterator Iterator[T], filter func(T) bool) Iterator[T] {
```

Comment:
I feel like `Filter` might be a better name. `iterator.Filter(myIterator, filter)`
```suggestion
// Filter returns an iterator that skips the elements in [it] that return true
// from [filter].
func Filter[T any](it Iterator[T], filter func(T) bool) Iterator[T] {
```